design/BufferOperations.md

```diff
@@ -152,25 +152,25 @@ stagingVertexBuffer.mapWriteAsync().then((stagingData) => {
 });
 ```
 
-### Updating data to an existing buffer (like webgl.bufferSubData)
+### Updating data to an existing buffer (like `gl.bufferSubData`)
 
-```
+```js
 function bufferSubData(device, destBuffer, destOffset, srcArrayBuffer) {
-    const byteCount = srcArrayBuffer.byteLength;
-    const (srcBuffer, mapping) = device.createBufferMapped({
-        size: byteCount,
-        usage: GPUBufferUsage.TRANSFER_SRC,
-    });
-    (new Uint8Array(mapping)).set(new Uint8Array(srcArrayBuffer)); // memcpy
-    srcBuffer.unmap();
-
-    const enc = device.createCommandEncoder({});
-    enc.copyBufferToBuffer(srcBuffer, 0, destBuffer, destOffset, byteCount);
-    const cb = enc.finish();
-    const q = device.getQueue();
-    q.submit([cb]);
-
-    srcBuffer.destroy();
+  const byteCount = srcArrayBuffer.byteLength;
+  const [srcBuffer, arrayBuffer] = device.createBufferMapped({
```

Comment:
Same fix below in AutoRingBuffer even if the rest of the code there hasn't been tested yet

Reply:
Done.